plugins/plugin-testing-java/plugin-testing-classpath/che-plugin-testing-classpath-maven-server/src/main/java/org/eclipse/che/plugin/testing/classpath/maven/server/MavenTestClasspathProvider.java

```diff
@@ -57,35 +64,55 @@ public String getProjectType() {
         return "maven";
     }
 
-    private boolean buildClasspath(String projectPath) throws IOException, InterruptedException {
-        final CommandLine commandLineClassPath = new CommandLine("mvn", "clean", "dependency:build-classpath",
-                "-Dmdep.outputFile=target/test.classpath.maven");
-        Process processBuildClassPath = new ProcessBuilder().redirectErrorStream(true).directory(new File(projectPath))
-                .command(commandLineClassPath.toShellCommand()).start();
-        ProcessUtil.process(processBuildClassPath, LineConsumer.DEV_NULL, LineConsumer.DEV_NULL);
-        processBuildClassPath.waitFor();
-        final CommandLine commandLineTestCompile = new CommandLine("mvn", "test-compile");
-        Process processTestCompile = new ProcessBuilder().redirectErrorStream(true).directory(new File(projectPath))
-                .command(commandLineTestCompile.toShellCommand()).start();
-        ProcessUtil.process(processTestCompile, LineConsumer.DEV_NULL, LineConsumer.DEV_NULL);
-        return processTestCompile.waitFor() == 0;
-
+    private Stream<ClasspathEntryDto> toResolvedClassPath(Stream<ClasspathEntryDto> rawClasspath) {
+        return rawClasspath.flatMap(dto -> {
+            if (dto.getEntryKind() == IClasspathEntry.CPE_CONTAINER) {
+                return toResolvedClassPath(dto.getExpandedEntries().stream());
+            } else {
+                return Stream.of(dto);
+            }
+        });
     }
 
-    private List<URL> getProjectClasspath(String projectPath) throws IOException {
-        List<URL> classUrls = new ArrayList<>();
-        File cpFile = Paths.get(projectPath, "target", "test.classpath.maven").toFile();
-        FileReader fileReader = new FileReader(cpFile);
-        BufferedReader bufferedReader = new BufferedReader(fileReader);
-        String line = bufferedReader.readLine();
-        String[] paths = line.split(":");
-        for (String path : paths) {
-            classUrls.add(new File(path).toURI().toURL());
-        }
-        bufferedReader.close();
-        fileReader.close();
-        classUrls.add(Paths.get(projectPath, "target", "classes").toUri().toURL());
-        classUrls.add(Paths.get(projectPath, "target", "test-classes").toUri().toURL());
-        return classUrls;
+    private Stream<URL> getProjectClasspath(String projectPath) throws JavaModelException {
```

Comment:
That will be, surely, but is it required *now*, or could I push the tests in a next pull request ?